Correcting file name for groups.uml

mlep opened the pull request:
Following message at https://discourse.jabref.org/t/github-documentation-uml-diagram-link-broken/2048 , fixing the file name.

Note: several links are in the form https://github.com/JabRef/jabref/tree/ec47f2138b0550a4622872d455902443cd56d9cc/docs/Groups.uml
Why not something like  https://github.com/JabRef/jabref/blob/master/docs/Groups.uml ?

Siedlerchr commented:
@mlep  Thanks for updating. The links with tree... are always refering to a specific commit in time. In most cases it makes sense to refer to a specific version as later versions might have change it.
And I think when you paste a link to tree/master it sometimes get replaced with the commit speciciv version automatically.
But for the documentation it's not necessary

koppor commented on docs/README.md:
```diff
@@ -26,7 +26,8 @@ Without jlink, it is not possible to generate a fat jar any more. During develop
 
-UML diagram showing aspects of groups: [Groups.uml](https://github.com/JabRef/jabref/tree/ec47f2138b0550a4622872d455902443cd56d9cc/docs/Gropus.uml).
+
+diagram showing aspects of groups: [Groups.uml](https://github.com/JabRef/jabref/tree/ec47f2138b0550a4622872d455902443cd56d9cc/docs/Groups.uml).
```
```suggestion
diagram showing aspects of groups: [Groups.uml](Groups.uml).
```

Maybe Gitbook auto corrects the link? :-\. I always use the direct links.

mlep commented:
@Siedlerchr thank you for the explanation. I have now simplified all the links with tree/ .
@koppor I have integrated a shorter link, as you suggested.
Additionnally, I have updated the section about Decision records.
